map: Дельта на низкой орбите

pilygun opened the pull request:
## Что этот ПР делает

Добавляет новую карту в папку ивент. 3х уровневая мульти Z дельта. 
1. Уровень лаваленда
2. Пустое пространство 
3. Собственно дельта

Большая чать мапы доделана, осталось отшлифовать моменты.
Так же огромная благодарность Вове, ибо он исправил проблемы с кодом и мапой вроде все даже работает.

## Почему это хорошо для игры

Свежий взгляд на старую мапу.
<img width="1216" height="960" alt="image" src="https://github.com/user-attachments/assets/7cb8c628-2037-47c3-bb70-da6881e66782" />


## Демонстрация изменений

<img width="1180" height="1012" alt="image" src="https://github.com/user-attachments/assets/f62182f4-2a12-41c7-8547-ece76c962439" />


## Тестирование

Локалка.

vovakons committed:
event map datum for delta low orbit

pilygun committed:
Дельта на низкой орбите

pilygun committed:
Дельта на низкой орбите

vovakons commented:
<img width="523" height="85" alt="image" src="https://github.com/user-attachments/assets/01d744d8-d0fd-493a-b63f-28aed03a784d" />

pilygun committed:
+КУЧА СПРАЙТОВ И БАГОВ

pilygun committed:
Чапка пипсов

pilygun committed:
приколы

pilygun committed:
огромное количество правок

pilygun committed:
Огомное коичество фиксов + батник для фиксов от родена

pilygun committed:
ебаная решетка ЛИКВИДИРОВАННА

dageavtobusnick commented on code/controllers/subsystem/mapping.dm:
```diff
@@ -164,7 +164,8 @@ SUBSYSTEM_DEF(mapping)
 		var/lavaland_setup_timer = start_watch()
-		seedRuins(list(level_name_to_num(MINING)), CONFIG_GET(number/lavaland_budget), /area/lavaland/surface/outdoors/unexplored, GLOB.lava_ruins_templates)
+		if(exists_level_name(MINING))
+			seedRuins(list(level_name_to_num(MINING)), CONFIG_GET(number/lavaland_budget), /area/lavaland/surface/outdoors/unexplored, GLOB.lava_ruins_templates)
```
А разве нижний Z уровень, заменяющий лаваленд, не должен заполнятся руинами?

dageavtobusnick commented on code/modules/lavaland/lavaland_theme.dm:
```diff
@@ -30,6 +30,8 @@
 	lavaland_type = LAVALAND_TYPE_LAVA
 
 /datum/lavaland_theme/lava/setup()
+	if(!exists_level_name(MINING))
```
Так же с этим всем.

dageavtobusnick reviewed (commented):
Для тестмерджа пойдет. Но вопросы с генерацией нужно будет решить

pilygun committed:
Фиксы стекл (они пропали да)

ROdenFL committed:
Merge branch 'master2200' into lavaland-gen-fix

pilygun committed:
фиксы стекл

dageavtobusnick committed:
Merge branch 'master220' into lavaland-gen-fix

littleboobs committed:
Merge branch 'master220' into lavaland-gen-fix

pilygun committed:
Фиксы мапинга

pilygun committed:
Merge branch 'lavaland-gen-fix' of https://github.com/pilygun/Paradise into lavaland-gen-fix

pilygun committed:
фиксы карты

pilygun committed:
мапы фиксы

vovakons committed:
multiz explosion scale

vovakons committed:
Merge branch 'lavaland-gen-fix' of github.com:pilygun/Paradise into lavaland-gen-fix

littleboobs committed:
Merge branch 'master220' into lavaland-gen-fix

dageavtobusnick commented on code/modules/map_fluff/event.dm:
```diff
@@ -53,3 +53,22 @@
+	english_station_name = "NSS Kerberos (low orbit)"
+	station_short = "Керберос"
+	dock_name = "АКН Трурль"
+	company_name = "НаноТрейзен"
```
```suggestion
	company_name = "Нанотрейзен"
```

dageavtobusnick committed:
Update code/modules/map_fluff/event.dm

pilygun committed:
Добавление спейс подов и фиксы

vovakons committed:
survival pod fix for low orbit delta

Добавлена проверка турфа для капсулы, теперь можно кидать и
разворачивать на турфе лаваленда, даже если это уровень станции.
Добавлено создание потолка для капсулы на мульти-Z лаваленде.

vovakons committed:
lavaland ruins generation for delta low orbit

Добавлен трейт для замены Z уровня лаваленда станционным уровнем.
Теперь нижний уровень карты delta на низкой орбите генерирует руины.

vovakons committed:
Merge branch 'master220' into lavaland-gen-fix

vovakons committed:
linter fixes

littleboobs commented on tools/mapmerge2/duplicate_check.py:
Этот скрипт не нужен, его надо удалить, он бесполезен с маплинтом. В репозитории есть нормальная и рабочая версия нахождения коллизий, если кому-то понадобится. check_wallbump_collisions.py

littleboobs commented on tools/mapmerge2/Found_Duplicates.bat:
Батник тоже.

littleboobs commented on code/datums/helper_datums/map_template.dm:
```diff
@@ -58,6 +58,15 @@
 			stack_trace("One of the late setup corners is bust")
 		if(ST_bot_left == null || ST_top_right == null)
 			stack_trace("One of the smoothing corners is bust")
+		if(roof_type)
+			var/turf/above_turf = locate(placement.x, placement.y, placement.z + 1)
+			if(above_turf) //exists above level
+				for(var/x = min_x; x <= max_x; ++x)
+					for(var/y = min_y; y <= max_y; ++y)
+						var/turf/turf = locate(x, y, placement.z + 1)
+						turf.ChangeTurf(roof_type)
```
МЕГА БРУХ

pilygun committed:
фиксы

pilygun committed:
Прости роден

dageavtobusnick commented:
один тестмердж придется потерпеть. Опять изменение путей.

dageavtobusnick committed:
Merge branch 'master220' into lavaland-gen-fix

littleboobs commented:
Можешь решить мержконфликт воспользовавшись скриптом. Просто перетащи инструкцию (tools\UpdatePaths\Scripts\7811_landmarks_refactor.txt) на батник скрипта.

vovakons committed:
Merge branch 'master220' into lavaland-gen-fix

vovakons committed:
Merge branch 'lavaland-gen-fix' of github.com:pilygun/Paradise into lavaland-gen-fix

vovakons committed:
fix delta low orbit map file

vovakons committed:
remove pregenerated ruins from delta low orbit

vovakons committed:
revert evil_santa.dmmm

dageavtobusnick committed:
Merge branch 'master220' into lavaland-gen-fix

pilygun committed:
Merge branch 'ss220-space:master220' into lavaland-gen-fix

Pelmenik123 commented:
тааак что тут за проблемы остались

pilygun committed:
Merge remote-tracking branch 'upstream/master220' into lavaland-gen-fix

pilygun committed:
фиксы

pilygun committed:
фикс

pilygun committed:
—

pilygun committed:
11

pilygun committed:
Добавлена крыша для всех руин

pilygun committed:
Ебаные стекла

pilygun committed:
фиксики

pilygun committed:
фиксы

littleboobs commented:
/gemini summary

littleboobs commented:
/gemini review

pilygun commented:
Слишком сложный проект для такого рукажопа как я.
Сам не вывожу,  скинуть не на кого. Чинить после каждого рефактора заебался, а как фиксануть генерацию крыш не ебу.
Бонусом огромная нагрузка на сервер из за атмоса, лаги даже на локалке, да и аудитории не то что бы сильно понравилась мапа на тестах. 
Так что смысла пинать труп более не вижу.
